open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/impl/VRPConsentRequestHandler.java

```diff
@@ -0,0 +1,222 @@
+                                ConsentExtensionConstants.VRP));
+                consentManageData.setResponseStatus(ResponseStatus.OK);
+            } catch (ConsentManagementException | ParseException e) {
+                log.error(ErrorConstants.INVALID_CLIENT_ID_MATCH, e);
+                throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
+                        ErrorConstants.ACC_INITIATION_RETRIEVAL_ERROR);
+            }
+        } else {
+            log.error(ErrorConstants.INVALID_CONSENT_ID);
+            throw new ConsentException(ResponseStatus.BAD_REQUEST, ErrorConstants.INVALID_CONSENT_ID);
+        }
+    }
+
+    /**
+     * Handles the DELETE request for revoking or deleting a consent.
+     *
+     * @param consentManageData Object containing request details
+     */
+    @Override
+    public void handleConsentManageDelete(ConsentManageData consentManageData) {
+
+        ConsentManageUtil.handleConsentManageDelete(consentManageData);
+    }
+
+    /**
+     * Method to handle Variable Recurring Payment POST requests.
+     * This private method processes Variable Recurring Payment POST requests, creating a new consent
+     * based on the provided request payload. It performs the following actions:
+     * - Creates a DetailedConsentResource representing the consent initiation.
+     * - Stores consent attributes, including the idempotency key.
+     * - Constructs the response payload containing initiation details and sets appropriate headers.
+     * - Sets the response status to Created.
+     *
+     * @param consentManageData Object containing request details, including client ID, request payload, headers.
+     * @param request Object
+     */
+    public void handlePaymentPost(ConsentManageData consentManageData, Object request)
+            throws ConsentManagementException {
+
+        if (request instanceof JSONObject) {
+            JSONObject requestObject = (JSONObject) request;
+
+            // Create a ConsentResource representing the requested consent
+            ConsentResource requestedConsent = new ConsentResource(consentManageData.getClientId(),
+                    requestObject.toJSONString(), ConsentExtensionConstants.VRP,
+                    ConsentExtensionConstants.AWAITING_AUTH_STATUS);
+
+            // Create the consent
+            DetailedConsentResource createdConsent = ConsentServiceUtil.getConsentService()
+                    .createAuthorizableConsent(requestedConsent, null,
+                            CREATED_STATUS, AUTH_TYPE_AUTHORIZATION, true);
+
+            //Set consent attributes for storing
+            Map<String, String> consentAttributes = new HashMap<>();
+            consentAttributes.put(ConsentExtensionConstants.IDEMPOTENCY_KEY, consentManageData.getHeaders()
+                    .get(ConsentExtensionConstants.X_IDEMPOTENCY_KEY));
+
+
+            JSONObject response = (JSONObject) request;
```

Comment:
Shall we remove this as this is redundant [1]

[1] https://github.com/wso2/financial-open-banking/pull/37/files#diff-616d670d8ef90047e4418fb2539c9cf2bc1042598f100d29185c91d006529d8bR163

Reply:
Fixed in [df474ab]